File: ui/src/app/configs/common/regex.patterns.ts
export class RegexPatterns {
    static kebabSnakeCase = /[A-Z]{2,}(?=[A-Z][a-z]+[0-9]*|\b)|[A-Z]?[a-z]+[0-9]*|[A-Z]|[0-9]+/g
    static camelCase = /[^a-zA-Z0-9]+(.)/g
    static pascalCase = /\w\S*/g
    static spaceCamel = /([A-Z]+)/g
}

File: src/Core/Constants/RegexPatterns.cs
using System.Text.RegularExpressions;

namespace Fwks.Core.Constants;

public static partial class RegexPatterns
{
    [GeneratedRegex("(?<!^)([A-Z][a-z]|(?<=[a-z])[A-Z])")]
    public static partial Regex TitleCase();

    [GeneratedRegex("(?:^|_| +)(.)")]
    public static partial Regex PascalCase();

    [GeneratedRegex("([\\p{Lu}]+)([\\p{Lu}][\\p{Ll}])")]
    public static partial Regex SlugSnakeStepOne();

    [GeneratedRegex("([\\p{Ll}\\d])([\\p{Lu}])")]
    public static partial Regex SlugSnakeStepTwo();

    [GeneratedRegex("[-\\s]")]
    public static partial Regex SlugSnakeStepThree();

    [GeneratedRegex("(\\d*)(b|kb|mb|gb|tb|){1}$")]
    public static partial Regex ByteTransform();
}